### PR TITLE
fix(unfied_sql): service and host states are filled at startup with r…

### PR DESCRIPTION
**Pull request by jean-christophe81**
## Description

**Fixes** # (issue)

real_state is copied in state in services and hosts tables at restart even if real_state is null

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [X] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/broker/start-stop.robot [hunk 1]
```diff
@@ -83,6 +87,47 @@ BSSU5
...
+    [Tags]    broker    start-stop    unified_sql
+    Config Broker    central
+    Config Broker    rrd
+    Config BBDO3    nbEngine=1
+    Config Engine    ${1}    ${50}    ${20}
+    Config Broker Sql Output    central    unified_sql
+
+    Clear Db    services
+    Clear Db    hosts
+    ${start}=    Get Current Date
```
Comment (thearchit3ct) [inline tests/broker/start-stop.robot:101]:
```suggestion
    ${start}    Get Current Date
```

File: tests/broker/start-stop.robot [hunk 2]
```diff
@@ -83,6 +87,47 @@ BSSU5
...
+
+    Start Engine
+    Start Broker
+
+    # wait engine start
+    ${content}=    Create List    INITIAL SERVICE STATE: host_50;service_1000;    check_for_external_commands()
+    ${result}=    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
```
Comment (thearchit3ct) [inline tests/broker/start-stop.robot:108]:
```suggestion
    ${result}    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
```

File: tests/broker/start-stop.robot [hunk 3]
```diff
@@ -83,6 +87,47 @@ BSSU5
...
+    Should Be True
+    ...    ${result}
+    ...    msg=An Initial service state on (host_50,service_1000) should be raised before we can start our external commands.
```
Comment (thearchit3ct) [inline tests/broker/start-stop.robot:111]:
```suggestion
    ...    An Initial service state on (host_50,service_1000) should be raised before we can start our external commands.
```

File: tests/broker/start-stop.robot [hunk 4]
```diff
@@ -83,6 +87,47 @@ BSSU5
...
+
+    # restart central broker
+    Stop Broker
+    Start Broker
+
+    Connect To Database    pymysql    ${DBName}    ${DBUser}    ${DBPass}    ${DBHost}    ${DBPort}
+
+    FOR    ${index}    IN RANGE    30
+        Sleep    1
+        ${output}=    Query    SELECT state FROM services WHERE enabled=1 AND state IS NULL
```
Comment (thearchit3ct) [inline tests/broker/start-stop.robot:121]:
```suggestion
        ${output}    Query    SELECT state FROM services WHERE enabled=1 AND state IS NULL
```

File: tests/broker/start-stop.robot [hunk 5]
```diff
@@ -83,6 +87,47 @@ BSSU5
     Broker Config Output Remove    central    centreon-broker-master-rrd    host
     Repeat Keyword    5 times    Start Stop Instance    1s
 
+START_STOP_CBD
+    [Documentation]    restart cbd with unified_sql services state must not be null after restartno deadlock
```
Comment (bouda1) [inline tests/broker/start-stop.robot:91]: There is a typo in this documentation.

**Review by bouda1 — CHANGES_REQUESTED** [review]

**Review by thearchit3ct — APPROVED** [review]

**Pull request merged**